src/main/java/duke/main/Parser.java

```diff
@@ -0,0 +1,51 @@
+package duke.main;
+
+import duke.commands.ByeCommand;
+import duke.commands.Command;
+import duke.commands.DeadlineTaskCommand;
+import duke.commands.DeleteCommand;
+import duke.commands.EventTaskCommand;
+import duke.commands.FindCommand;
+import duke.commands.ListCommand;
+import duke.commands.MarkCommand;
+import duke.commands.ToDoTaskCommand;
+import duke.commands.UnmarkCommand;
+import duke.exception.DukeException;
+
+/**
+ * A parser that parses user input to the commands
+ */
+public class Parser {
+
+    /**
+     * Returns the correct command based on user input
+     *
+     * @param text The text received from user
+     * @throws DukeException If user types in invalid or incomplete commands
+     */
+    static Command parse(String text) throws DukeException {
+        String command = text.split(" ")[0];
+        switch (CommandManager.valueOf(command)) {
+        case list:
```

Comment:
Great use of switch-case. Following the indentation guide according to the Java coding standards!